### PR TITLE
🎨 Check for zero credits (if pricing unit cost is greater than 0)

### PR DESCRIPTION
**Pull request by matusdrobuliak66**
## What do these changes do?
- 🎨 Check for zero credits (if pricing unit cost is greater than 0)

## Related issue/s
- relates https://github.com/ITISFoundation/osparc-issues/issues/1437
- closes https://github.com/ITISFoundation/osparc-issues/issues/1438

## Dev-ops checklist

- [ ] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

File: packages/models-library/src/models_library/director_v2.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+
+class PricingInfo(BaseModel):
+    pricing_plan_id: PricingPlanId
+    pricing_unit_id: PricingUnitId
+    pricing_unit_cost_id: PricingUnitCostId
+    pricing_unit_cost: Decimal
+
+    class Config:
+        schema_extra: ClassVar[dict[str, Any]] = {
```
Comment (sanderegg) [inline packages/models-library/src/models_library/director_v2.py:16]: hmm why is this module called ```director_v2.py``` is there not something wrong there?

File: services/director-v2/src/simcore_service_director_v2/core/errors.py [hunk 2]
```diff
@@ -114,6 +116,13 @@ class ComputationalTaskNotFoundError(PydanticErrorMixin, DirectorError):
     msg_template = "Computational task {node_id} not found"
 
 
+class WalletNotEnoughCreditsError(DirectorError):
+    """Project not found error"""
+
+    def __init__(self, wallet_name: str, wallet_credit_amount: Decimal):
+        super().__init__(f"Wallet '{wallet_name}' has {wallet_credit_amount} credits.")
+
```
Comment (sanderegg) [inline services/director-v2/src/simcore_service_director_v2/core/errors.py:124]:
```suggestion
class WalletNotEnoughCreditsError(PydanticErrorMixin, DirectorError):
    msg_template = "Wallet '{wallet_name}' has {wallet_credit_amount} credits."
```

File: services/director-v2/tests/unit/with_dbs/test_api_route_computations.py [hunk 3]
```diff
@@ -384,7 +400,11 @@ async def test_create_computation(
 
 @pytest.fixture
 def wallet_info(faker: Faker) -> WalletInfo:
-    return WalletInfo(wallet_id=faker.pyint(), wallet_name=faker.name())
+    return WalletInfo(
+        wallet_id=faker.pyint(),
+        wallet_name=faker.name(),
+        wallet_credit_amount=Decimal(10),
```
Comment (sanderegg) [inline services/director-v2/tests/unit/with_dbs/test_api_route_computations.py:406]:
```suggestion
        wallet_credit_amount=Decimal(faker.pyint(min_value=12, max_value=129312)),
```

File: services/web/server/src/simcore_service_webserver/projects/_nodes_handlers.py [hunk 4]
```diff
@@ -303,21 +307,16 @@ async def start_node(request: web.Request) -> web.Response:
...
+        user_id=req_ctx.user_id,
+        project_id=path_params.project_id,
+        node_id=path_params.node_id,
+    )
 
-    except ProjectStartsTooManyDynamicNodesError as exc:
-        raise web.HTTPConflict(reason=f"{exc}") from exc
-    except ClustersKeeperNotAvailableError as exc:
-        raise web.HTTPServiceUnavailable(reason=f"{exc}") from exc
+    raise web.HTTPNoContent(content_type=MIMETYPE_APPLICATION_JSON)
```
Comment (sanderegg) [inline services/web/server/src/simcore_service_webserver/projects/_nodes_handlers.py:320]: just wondering if these are generic node exception errors or only start ones, but ok.

File: services/web/server/src/simcore_service_webserver/projects/_states_handlers.py [hunk 5]
```diff
@@ -159,6 +159,7 @@ async def open_project(request: web.Request) -> web.Response:
 
         return envelope_json_response(project)
 
+    # MD: this might be causing troubles?
```
Comment (sanderegg) [inline services/web/server/src/simcore_service_webserver/projects/_states_handlers.py:162]: actually now I wonder also. the project is only deemed closed again if there is a DrectorServiceError, I wonder if there are other problems whether the project is also properly closed again?

**Review by sanderegg — COMMENTED** [review]
looking promising, but there are a few things I'd like to discuss. Thanks!

**Review by GitHK — APPROVED** [review]
look ok to me

**Review by sanderegg — APPROVED** [review]
thanks!

**Pull request merged**